cmake/AwsCFlags.cmake

```diff
@@ -231,7 +231,6 @@ function(aws_set_common_properties target)
 
         # try to check whether compiler supports LTO/IPO
         if (POLICY CMP0069)
```

Comment:
I know we're not trying to rewrite the world, but since this touches a POLICY ... we can replace this ~if (POLICY CMP0069)~ block with simpler code
```
        # check whether compiler supports LTO/IPO
        include(CheckIPOSupported)
        check_ipo_supported(RESULT ipo_supported)
        if (ipo_supported)
            message(STATUS "Enabling IPO/LTO for Release builds")
        else()
            message(STATUS "AWS_ENABLE_LTO is enabled, but cmake/compiler does not support it, disabling")
            set(_ENABLE_LTO_EXPR OFF)
        endif()
```
see: https://cmake.org/cmake/help/v3.9/policy/CMP0069.html